app/models/library_entry.rb

```diff
@@ -218,6 +211,18 @@ def kind
     end
   end
 
+  after_save if: :progress_changed? do
```

Comment:
should this really be an `after_save`? Might be better to do `after_commit on: :update` and then wrap the statement in an `if progress_changed?` Only saying this because `after_save` deals with both `create` and `update` and you already have a create below.